doc/nrf/ug_dev_model.rst

```diff
@@ -0,0 +1,514 @@
+
+   - ``origin`` usually points to the user's personal copy of the repository
+   - ``ncs`` is used to point to the |NCS| repository
+   - ``upstream`` typically points to the upstream repository, when applicable
+
+If you want to create a `GitHub fork`_ follow the steps below:
+
+#. Create a `GitHub fork`_ using the "Fork" button in the GitHub UI
+#. Add the newly created remote repository as a Git remote::
+
+     cd ncs/{folder}
+     # Rename the default remote from 'origin' to 'ncs'
+     git remote rename origin ncs
+     git remote add origin https://github.com/{username}/{repo}.git
+
+   For example, if you were creating a fork of the `fw-nrfconnect-nrf`_
+   repository for GitHub user ``foo``::
+
+     cd ncs/nrf
+     git remote rename origin ncs
+     git remote add origin https://github.com/foo/fw-nrfconnect-nrf.git
+
+   If you were to fork an OSS repository instead, which itself is already a fork
+   of the original upstream project::
+
+     cd ncs/zephyr
+     git remote rename origin ncs
+     git remote add origin https://github.com/foo/fw-nrfconnect-zephyr.git
+     git remote add upstream https://github.com/zephyrproject-rtos/zephyr.git
+
+  That way you would actually have 3 remotes, each pointing to the relevant copy
+  of the zephyr codebase:
+
+    * ``origin`` pointing to your own fork of ``fw-nrfconnect-zephyr``
+    * ``ncs`` pointing to the |NCS| `fw-nrfconnect-zephyr`_
+    * ``upstream`` pointing to the upstream `official Zephyr repository`_
+
+If you want to create a regular fork, you would follow the exact same
+steps as above, but the actual repository would need to be created by you
+beforehand, instead of pushing the "Fork" button. Also, and since a GitHub fork
+automatically initializes the forked repo with the exact same contents as the
+original one, you would need to push the contents yourself::
+
+  cd ncs/{folder}
+  # Rename the default remote from 'origin' to 'ncs'
+  git remote rename origin ncs
+  git remote add origin https://github.com/{username}/{repo}.git
+  git push origin master
+
+Workflows
+=========
+
+Below you can find a few practical workflows that can be used by an application
+developer. Which one to choose will depend on the type of application, the
+timeframe to develop it and the need to update the |NCS| version used. All
+workflows are described under the following basic assumptions:
+
+- One or more applications are to be developed using the |NCS|
+- Additional board definitions might be required by the user
+- Additional libraries might be required by the user
+- The term "applicaiton" henceforth includes the application code and any board
+  definitions and libraries it requires
+- The application(s) will require updates of the |NCS| revision
+
+Worflow 1: Eschew Git and west
+------------------------------
+
+If you have your own version control tools, you might want to simply not use Git
+nor west at all, and instead rely on your own toolset. In that case, you will
```

Comment:
s/nor west/or west/

AFAIK, the proper use of "nor" is with "neither": "neither fish nor fowl"